crates/services/p2p/src/p2p_service/tests.rs

```diff
@@ -0,0 +1,1388 @@
+                    let msg_acceptance = to_message_acceptance(&acceptance);
+                    node_b.report_message_validation_result(&message_id, peer_id, msg_acceptance);
+                }
+            },
+            node_c_event = node_c.next_event() => {
+                if let Some(FuelP2PEvent::GossipsubMessage { message_id, peer_id, .. }) = node_c_event {
+                    let msg_acceptance = to_message_acceptance(&acceptance);
+                    node_c.report_message_validation_result(&message_id, peer_id, msg_acceptance);
+                }
+            },
+            _ = interval.tick() => {
+                let mut transactions = vec![];
+                for _ in 0..amount_of_msgs_per_second {
+                    let random_tx =
+                        TransactionBuilder::script(rand::thread_rng().gen::<[u8; 32]>().to_vec(), rand::thread_rng().gen::<[u8; 32]>().to_vec()).finalize_as_transaction();
+
+                    transactions.push(random_tx.clone());
+                    let random_tx = GossipsubBroadcastRequest::NewTx(Arc::new(random_tx));
+
+                    match rand::thread_rng().gen_range(1..=3) {
+                        1 => {
+                            // Node A sends a Transaction
+                            let _ = node_a.publish_message(random_tx);
+
+                        },
+                        2 => {
+                            // Node B sends a Transaction
+                            let _ = node_b.publish_message(random_tx);
+
+                        },
+                        3 => {
+                            // Node C sends a Transaction
+                            let _ = node_c.publish_message(random_tx);
+                        },
+                        _ => unreachable!("Random number generator is broken")
+                    }
+                }
+
+                eprintln!("Node A WORLD VIEW");
+                eprintln!("B score: {:?}", node_a.get_peer_score(&node_b.local_peer_id).unwrap());
+                eprintln!("C score: {:?}", node_a.get_peer_score(&node_c.local_peer_id).unwrap());
+                eprintln!();
+
+                eprintln!("Node B WORLD VIEW");
+                eprintln!("A score: {:?}", node_b.get_peer_score(&node_a.local_peer_id).unwrap());
+                eprintln!("C score: {:?}", node_b.get_peer_score(&node_c.local_peer_id).unwrap());
+                eprintln!();
+
+                eprintln!("Node C WORLD VIEW");
+                eprintln!("A score: {:?}", node_c.get_peer_score(&node_a.local_peer_id).unwrap());
+                eprintln!("B score: {:?}", node_c.get_peer_score(&node_b.local_peer_id).unwrap());
+                eprintln!();
+
+                // never ending loop
+                // break;
+            }
+        }
+    }
+}
+
+// TODO: Move me before tests that use this function
+/// Reusable helper function for Broadcasting Gossipsub requests
+async fn gossipsub_broadcast(
```

Comment:
This method was updated to accept the new `GossipsubBroadcastRequest::Confirmations` variant.

Introduced `check_message_matches_request` as part of that (before it was hardcoded to look for a specific value).